src/actions/exchangeActions.js

```diff
@@ -376,11 +381,21 @@ export const getExchangeSupportedAssetsAction = (callback?: () => void) => {
       });
     }
 
-    const fetchSuccess: boolean = Array.isArray(assetsSymbols[0]) && Array.isArray(assetsSymbols[1]);
+    // if fetching failed and user can fall back to valid assets, no need to do anything
+    if (_exchangeSupportedAssets?.length && !fetchSuccess) return;
+
+    let fetchedAssetsSymbols: string[] = [];
+    if (fetchSuccess) {
+      fetchedAssetsSymbols = uniq(assetsSymbols[0].concat(assetsSymbols[1]));
+    } else if (fetchOneInchSuccess) {
+      fetchedAssetsSymbols = uniq(assetsSymbols[0]);
+    } else if (fetchUniswapSuccess) {
+      fetchedAssetsSymbols = uniq(assetsSymbols[1]);
+    }
 
-    const fetchedAssetsSymbols: string[] = fetchSuccess ? uniq(assetsSymbols[0].concat(assetsSymbols[1])) : [];
+    if (!fetchedAssetsSymbols.length) return;
```

Comment:
Maybe send this case Sentry? Just monitor cases when users fail to fetch and doesn't have any set, wdyt?